### PR TITLE
feat(source): embed protobuf well known types when using schema registry

### PR DESCRIPTION
**Pull request by xiangjinwu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

These 12 proto source files containing well known types shall be [bundled](https://github.com/protocolbuffers/protobuf/releases/download/v25.0/protoc-25.0-osx-aarch_64.zip) together with protoc / libprotobuf as they [may not be available](https://docs.confluent.io/platform/7.5/schema-registry/fundamentals/serdes-develop/serdes-protobuf.html#developer-notes:~:text=Do%20not%20upload%20Google/Protobuf%20dependencies%20into%20Schema%20Registry.) from schema registry.

## Checklist

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

Protobuf well known types can now be used without including them in schema registry.

### CODE REVIEW

File: ci/scripts/e2e-source-test.sh [hunk 1]
```diff
@@ -98,6 +98,10 @@ RUST_LOG="info,risingwave_stream=info,risingwave_batch=info,risingwave_storage=i
 cargo make ci-start ci-1cn-1fe
 python3 -m pip install requests protobuf confluent-kafka
 python3 e2e_test/schema_registry/pb.py "message_queue:29092" "http://message_queue:8081" "sr_pb_test" 20
+echo "make sure google/protobuf/source_context.proto is NOT in schema registry"
```
Comment (BugenZhao) [inline ci/scripts/e2e-source-test.sh:101]: This is really a good practice!

**Review by BugenZhao — COMMENTED** [review]
Does https://docs.rs/prost-types/latest/prost_types/ help?

**Comment by xiangjinwu** [comment]
> Does https://docs.rs/prost-types/latest/prost_types/ help?

It only contains generated rust code, rather than the raw proto source required by `SourceTreeDescriptorDatabase::build_file_descriptor_set`.

**Review by BugenZhao — APPROVED** [review]
LGTM!

**Pull request merged**